src/Mews.Fiscalization.Greece/RestClient.cs

```diff
@@ -48,75 +48,91 @@ internal RestClient(string userId, string subscriptionKey, AadeEnvironment envir
-
-            try
-            {
-                response = await HttpClient.SendAsync(requestMessage).ConfigureAwait(continueOnCapturedContext: false);
-            }
-            catch (Exception ex)
-            {
-                stopwatch.Stop();
-                Logger?.Info($"HTTP request failed after {stopwatch.ElapsedMilliseconds}ms.", new { HttpRequestDuration = stopwatch.ElapsedMilliseconds });
-
-                return BuildResponseDocWithErrors(SendInvoiceErrorCodes.TimeoutErrorCode, ex.Message, invoicesDoc.Invoices);
-            }
-
-            stopwatch.Stop();
-            Logger?.Info($"HTTP request finished in {stopwatch.ElapsedMilliseconds}ms.", new { HttpRequestDuration = stopwatch.ElapsedMilliseconds });
-
-            if (response.StatusCode == System.Net.HttpStatusCode.Unauthorized)
-            {
-                return BuildResponseDocWithErrors(SendInvoiceErrorCodes.ForbiddenErrorCode, "Authorization error", invoicesDoc.Invoices);
-            }
-            else if (response.StatusCode == System.Net.HttpStatusCode.InternalServerError)
-            {
-                return BuildResponseDocWithErrors(SendInvoiceErrorCodes.InternalServerErrorCode, "Internal server error", invoicesDoc.Invoices);
-            }
-
-            var responseContent = await response.Content.ReadAsStringAsync().ConfigureAwait(continueOnCapturedContext: false);
+                        var responseContent = await httpResponseMessage.Content.ReadAsStringAsync().ConfigureAwait(continueOnCapturedContext: false);
 
-            var responseDoc = XmlManipulator.Deserialize<ResponseDoc>(responseContent);
-            Logger?.Debug("Result received and successfully deserialized.", responseDoc);
+                        var responseDoc = XmlManipulator.Deserialize<ResponseDoc>(responseContent);
+                        Logger?.Debug("Result received and successfully deserialized.", responseDoc);
 
-            return responseDoc;
+                        return responseDoc;
+                    }
+                );
         }
 
-        internal async Task<bool> CheckUserCredentialsAsync()
+        internal Task<CheckUserCredentialsResult> CheckUserCredentialsAsync()
         {
-            HttpResponseMessage response;
             var queryString = HttpUtility.ParseQueryString(string.Empty);
+            queryString["mark"] = "mark";
+
+            var endpointUri = new Uri(BaseUri, $"{GetRequestDocsEndpointMethodName}?{queryString}");
+
+            return SendAsync(
+                sendFunc: 
+                    () => HttpClient.GetAsync(endpointUri),
+                buildErrorResponseFunc:
+                    (errorCode, errorMessage) => BuildCheckUserCredentialsResponseWithError(errorCode, errorMessage),
+                buildResponseFunc:
+                    httpResponseMessage => Task.FromResult(new CheckUserCredentialsResult(httpResponseMessage.StatusCode != System.Net.HttpStatusCode.Unauthorized))
+                );
```

Comment:
Same here. Just one indent level.
```suggestion
            return SendAsync(
                sendFunc: () => HttpClient.GetAsync(endpointUri),
                buildErrorResponseFunc: (errorCode, errorMessage) => BuildCheckUserCredentialsResponseWithError(errorCode, errorMessage),
                buildResponseFunc: httpResponseMessage => Task.FromResult(new CheckUserCredentialsResult(httpResponseMessage.StatusCode != System.Net.HttpStatusCode.Unauthorized))
            );
```